### DISCUSSION TITLE
Is it possible to provide a condition to a many to many relation?

### DISCUSSION THREAD

**Discussion by wadetandy** [prisma/prisma#2858] [Q&A] [answered]
I just started spiking out Prisma for a new project and am loving the experience so far! My question is about whether it's possible create a many-to-many relation that specifies some additional conditions as part of the table join.

Say I have models User and Group.  I also have a join table `GroupMembership`:

```prisma
model GroupMembership {
  userId String
  user    User @relation(fields: [userId], references: [id])
  groupId String
  group   Group @relation(fields: [groupId], references: [id])
  role String
}
```

You may notice the join table also has a `role` attribute. For our purposes say it can have a value of `admin` or `member`. I want to create two different relations on the `Group` model which take these into account:

```prisma
model Group {
  # other stuff
  admins User[] # restrict this to group memberships with role == admin
  members User[] # restrict to group memberships with role == member
}
```

Is this doable? Very straightfoward SQL to write, but can't find a reference to doing this in the docs.  

I'm using this with nexus + graphql, so I'm also open to other approaches for exposing this to my schema/resolvers if that's not something that can be done easily out of the box.

**Comment by ryands17** [comment, marked as answer]
Hey @wadetandy :wave: 
Unfortunately specifying conditions for relations directly in the schema is not possible at the moment. Could you create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for the above use case.

**Reply by wadetandy** [reply]
created https://github.com/prisma/prisma/issues/2881

**Comment by ludralph** [comment]
Hi @wadetandy,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
